### ISSUE TITLE
undefined: base64Decode

### ISSUE THREAD

**Issue by phreek1337** [SaturnsVoid/GoBot2#2]
Hey,

I've got some problems while compile the Server.go

`./Server.go:81: undefined: base64Decode                                         
./Server.go:83: undefined: base64Decode                                         
./Server.go:85: undefined: base64Decode                                         
./Server.go:87: undefined: base64Decode                                         
./Server.go:89: undefined: base64Decode                                         
./Server.go:90: undefined: base64Decode                                         
./Server.go:92: undefined: base64Decode                                         
./Server.go:93: undefined: base64Decode                                         
./Server.go:205: undefined: logUpdate                                           
./Server.go:209: undefined: createDir                                           
./Server.go:209: too many errors`

I can't get Work. Would be nice If you could help me Out.

**Comment by MShoaei** [comment]
> 
> 
> Hey,
> 
> I've got some problems while compile the Server.go
> 
> `./Server.go:81: undefined: base64Decode ./Server.go:83: undefined: base64Decode ./Server.go:85: undefined: base64Decode ./Server.go:87: undefined: base64Decode ./Server.go:89: undefined: base64Decode ./Server.go:90: undefined: base64Decode ./Server.go:92: undefined: base64Decode ./Server.go:93: undefined: base64Decode ./Server.go:205: undefined: logUpdate ./Server.go:209: undefined: createDir ./Server.go:209: too many errors`
> 
> I can't get Work. Would be nice If you could help me Out.

found the answer! cd into the console server directory and run `go build` this will produce "Console Server.exe" and use command line to run the executable

**Comment by MShoaei** [comment]
> 
> 
> after running go build on linux I got executable file for linux so I ran chmod +x on the file but when I launch it nothing happens did you managed to make it run @MShoaei

yes I managed to run the server flawlessly. what do u mean by "Nothing happens"? Do u have a database running? Does a banner print out in the terminal?

**Comment by vladimirpe00** [comment]
I was able to launch it and manage it from panel but the issue that keylogger only gathering the app the victim is using (testing it on my vm) but never catching keystrokes neither clipboard.

and any command sent to the victim went with successful output on panel but nothing happened on client side so I don't if you faced same issues

**Comment by MShoaei** [comment]
> 
> 
> I was able to launch it and manage it from panel but the issue that keylogger only gathering the app the victim is using (testing it on my vm) but never catching keystrokes neither clipboard.
> 
> and any command sent to the victim went with successful output on panel but nothing happened on client side so I don't if you faced same issues
> 
> UPDATE:
> 
> it seems when send any command from C&C to client the command not executed even with successful output, and it will kill the agent no sync to C&C anymore but it will stay in working the system

This build is quite buggy. many features aren't working and some features are just over kills and consume a lot of processing power. I suggest disabling the features and enabling them one at a time to find the broken spot. You can check out my version of this bot, too.

**Comment by vladimirpe00** [comment]
> > I was able to launch it and manage it from panel but the issue that keylogger only gathering the app the victim is using (testing it on my vm) but never catching keystrokes neither clipboard.
> > and any command sent to the victim went with successful output on panel but nothing happened on client side so I don't if you faced same issues
> > UPDATE:
> > it seems when send any command from C&C to client the command not executed even with successful output, and it will kill the agent no sync to C&C anymore but it will stay in working the system
> 
> This build is quite buggy. many features aren't working and some features are just over kills and consume a lot of processing power. I suggest disabling the features and enabling them one at a time to find the broken spot. You can check out my version of this bot, too.
> I haven't got the time to test the keylogger. I think you can test that yourself.

indeed what you said is 100% correct I didn't mentioned the consuming, besides I tried to send any command just to test stability it seems the client goes dead after 40min max of activity then you will lose access I will try to check features as you said.

**Comment by TormentedSoul666** [comment]
Take a quick look at the dependencies, it's only compatible with Windows...
